### PR TITLE
Added CoEVP Spackage

### PR DESCRIPTION
**Pull request by rspavel**
Added spackage for baseline (mpi, no other dependencies) CoEVP

### CODE REVIEW

File: var/spack/repos/builtin/packages/coevp/package.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    reimplement it as needed to test performance of new architectures,
+    programming models, etc.
+    Due to the size and complexity of the studied models, as well as
+    restrictions on distribution, the currently available LULESH proxy
+    application provides the coarse-scale model implementation and the ASPA
+    proxy application provides the adaptive sampling support."""
+
+    homepage = 'https://github.com/exmatex/CoEVP'
+
+    version('master', git='https://github.com/exmatex/CoEVP.git',
```
Comment (davydden) [inline var/spack/repos/builtin/packages/coevp/package.py]: call name it `develop`, see http://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison

Comment (rspavel) [reply var/spack/repos/builtin/packages/coevp/package.py]: Does that necessarily make sense in this case? CoEVP doesn't really follow a gitflow model and treats master as the end-all. And I notice spackages for projects with similar models (e.g. CoMD) also just refer to the "master" branch as "master"

Comment (davydden) [reply var/spack/repos/builtin/packages/coevp/package.py]: > Does that necessarily make sense in this case? 

yes

>  And I notice spackages for projects with similar models (e.g. CoMD) also just refer to the "master" branch as "master"

that's a mistake. The problem is, as soon as you have a release version, say, `x.y.z` and then somebody who use your package writes
```
depends_on(foo@x.y.z:)
```
then your `foo@master` aint gonna satisfy this dependency. So you are really advised to name the end-all as `develop`.

Comment (davydden) [reply var/spack/repos/builtin/packages/coevp/package.py]: p.s. there is a PR which makes `master` behave the same https://github.com/LLNL/spack/pull/1983 but it has not been merged.

**Comment by rspavel** [comment]
I still disagree with referring to the master branch as "develop" as this will lead to headaches as more projects with non-gitflow style development models are added to spack ("Install the develop spackage. No, not the development branch, the 'develop' spackage. You know, 'master'"), but made edit as per current spack practices.

**Comment by davydden** [comment]
>  I still disagree with referring to the master branch as "develop"

as i said, there is a PR which improves this https://github.com/LLNL/spack/pull/1983 but it has not been merged. It's not up to me to merge it.

> non-gitflow style development models

what model is adopted by each project is irrelevant here. It's only about version comparison and least work to specify constraints on dependencies. So far there is a single name for a version is is meant to be greater than any release, see Package Guide and the rationale behind it http://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison

**Comment by junghans** [comment]
Rebase on recent develop branch to fix CI issue.

File: var/spack/repos/builtin/packages/coevp/package.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            branch='master')
+
+    variant('mpi', default=True, description='Build with MPI Support')
+
+    depends_on('mpi', when='+mpi')
+
+    def makefile_args(self):
+        spec = self.spec
+        args = []
+        args.append('FLANN=no')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/coevp/package.py]: There is a `flann` in spack.

File: var/spack/repos/builtin/packages/coevp/package.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        args.append('REDIS=no')
+        args.append('SILO=no')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/coevp/package.py]: There is a `silo` in spack.

File: var/spack/repos/builtin/packages/coevp/package.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    homepage = 'https://github.com/exmatex/CoEVP'
+
+    version('develop', git='https://github.com/exmatex/CoEVP.git',
+            branch='master')
+
+    variant('mpi', default=True, description='Build with MPI Support')
+
+    depends_on('mpi', when='+mpi')
+
+    def makefile_args(self):
```
Comment (junghans) [inline var/spack/repos/builtin/packages/coevp/package.py]: What is `makefile_args()`?

**Pull request merged**